src/functions.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+function displayDeity(array $deity): string 
+
+{
+
+         if (array_key_exists('name', $deity) 
+        && array_key_exists('mythology', $deity) 
+        && array_key_exists('association', $deity) 
+        && array_key_exists('objects', $deity)) 
+        
+        {
+	return '<h2>' . $deity['name'] . '</h2>' 
+        . '<p>' . $deity['name'] . ' is a deity from ' . $deity['mythology'] . ' mythology,</p>'
+        . '<p>is associated with ' . $deity['association'] . ',</p>'
+        . '<p> and is usually portrayed holding ' . $deity['objects'] . '.</p>';
+        } 
+        
+        else 
+        { 
+        throw new Exception('Invalid data'); 
```

Comment:
you could do with a more informative error message (because it is supposed to be useful for you to diagnose issues by reading it)